MPDP-710 Cookies consent

rtasalem opened the pull request:
https://eaflood.atlassian.net/browse/MPDP-710

johnwatson484 reviewed (commented):
If I navigate to cookies, I get a banner saying that I've updated my cookie policy before I've set it.

<img width="1064" height="502" alt="image" src="https://github.com/user-attachments/assets/e7033a8c-6ec2-4fd3-a825-7fd38671872b" />

johnwatson484 commented on src/config/config.js:
```diff
@@ -156,6 +156,62 @@ export const config = convict({
+      format: String,
+      default: 'fcp_mpdp_cookie_policy',
+    },
+    nameAuth: {
+      doc: 'Name of cookie authentication',
+      format: String,
+      default: 'fcp_mpdp_auth'
+    },
+    nameSession: {
+      doc: 'Name of cookie session',
+      format: String,
+      default: 'fcp_mpdp_session'
+    },
```
These two properties aren't used anywhere?

rtasalem replied:
Both have been removed.

johnwatson484 commented on src/config/config.js:
```diff
@@ -156,6 +156,62 @@ export const config = convict({
+      default: isProduction
+    }
+  },
+  cookieConfig: {
+    ttl: {
+      doc: 'Time to live for cookie policy (ms)',
+      format: 'nat',
+      default: 1000 * 60 * 60 * 24 * 365
+    }
```
Does this need to be separate from the other two cookie config sections?

Could we have a single `cookie` property with child objects if needed?  Rather than multiple top level values

rtasalem replied:
I've put everything under one `cookie` property and then since there's some overlap I've put in child objects for `policy` and `config`.

johnwatson484 reviewed (commented):
The cookies banner will keep appearing even when I accept or reject the policy

johnwatson484 commented:
The cookies page always shows my cookie status as not accepted, even when I have.

<img width="516" height="316" alt="image" src="https://github.com/user-attachments/assets/ff55f963-b7b7-490b-9cd6-f2f2dd96ea15" />

rtasalem commented:
> If I navigate to cookies, I get a banner saying that I've updated my cookie policy before I've set it.
> 
> <img alt="image" width="1064" height="502" src="https://private-user-images.githubusercontent.com/42574977/496054800-e7033a8c-6ec2-4fd3-a825-7fd38671872b.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTkzMTY4MjYsIm5iZiI6MTc1OTMxNjUyNiwicGF0aCI6Ii80MjU3NDk3Ny80OTYwNTQ4MDAtZTcwMzNhOGMtNmVjMi00ZmQzLWE4MjUtN2ZkMzg2NzE4NzJiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTEwMDElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMDAxVDExMDIwNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTVlMjUwM2Q0ZjllYjNiNzkxMDcyMmUzNDAxODc4ZDNhNWY1NTgyNWU3Zjc5ODUzOTFlZmE2ZjNjYzU3MDFhODAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.REskq_r-k92iVaVjZ5xbqGi0emXi--UT4e8MKj7fItY">

This is now fixed. Issue was the order I was passing the arguments in for the `cookiesModel`.

rtasalem commented:
> The cookies page always shows my cookie status as not accepted, even when I have.
> 
> <img alt="image" width="516" height="316" src="https://private-user-images.githubusercontent.com/42574977/496058136-ff55f963-b7b7-490b-9cd6-f2f2dd96ea15.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTkzMTY4MjYsIm5iZiI6MTc1OTMxNjUyNiwicGF0aCI6Ii80MjU3NDk3Ny80OTYwNTgxMzYtZmY1NWY5NjMtYjdiNy00OTBiLTljZDYtZjJmMmRkOTZlYTE1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTEwMDElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMDAxVDExMDIwNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTYzOWM2NmU4YjM4OGFhNmIwYjE4MzlhMmY5NTUzMDA2MjMxYWU5YzFjODVkOTVhYTJhYmI3OTRjMzdiYjZmYTgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.vxy_XcaD99uNbqkOL9meQMq9yCLgTfS-F03wqzN59yU">

This is now fixed and should mimic the current live service where if you click accept cookies and navigate to the cookies page the default option selected should be `Yes` and vice versa for if you reject the cookies.

johnwatson484 commented on src/config/config.js:
```diff
@@ -156,6 +157,52 @@ export const config = convict({
+        format: String,
+        default: 'base64json'
+      },
+      isSameSite: {
+        doc: 'Check if site is the same',
+        format: String,
+        default: 'Lax'
+      },
+      isSecure: {
+        doc: 'Check if secure',
+        format: Boolean,
+        default: isProduction
+      }
```
These two properties are duplicated.

rtasalem replied:
Will take out the `isSameSite` and `isSameSecure` from the top level `cookies` property so that they just exist within the `policy` child object.

johnwatson484 commented on src/plugins/cookies.js:
```diff
@@ -0,0 +1,39 @@
+          statusCode !== httpConstants.HTTP_STATUS_FORBIDDEN &&
+          statusCode !== httpConstants.HTTP_STATUS_BAD_REQUEST &&
+          statusCode !== httpConstants.HTTP_STATUS_INTERNAL_SERVER_ERROR &&
+          request.response.source.manager._context
```
This line needs amending.

rtasalem replied:
Already done. 👍🏽

rtasalem commented on src/config/config.js:
```diff
@@ -140,6 +140,7 @@ export const config = convict({
     endpoint: {
       doc: 'Endpoint for fcp-mpdp-backend',
       format: String,
+      nullable: true,
```
Have removed `nullable: true`.

rtasalem commented on src/routes/cookies.js:
```diff
@@ -12,6 +12,7 @@ export const cookies = [
         'cookies/policy',
         {
           pageTitle: 'Cookies',
+          crumb: request.plugins.crumb,
```
Gone.

rtasalem commented on src/routes/cookies.js:
```diff
@@ -48,6 +47,7 @@ export const cookies = [
           'cookies/policy',
           {
             pageTitle: 'Cookies',
+            crumb: request.plugins.crumb,
```
Also gone.

rtasalem commented on src/client/javascripts/cookies.js:
```diff
@@ -16,11 +16,15 @@ export default {
       xhr.open('POST', '/cookies', true)
+
       xhr.setRequestHeader('Content-Type', 'application/json')
+      xhr.setRequestHeader('X-CSRF-Token', crumb)
```
Removed.
